Add `CREATE VECTOR INDEX` command for node vector indexes (#800)
neo4j/docs-cypher: modules/ROOT/pages/indexes-for-vector-search.adoc

> | Usage | Procedure/Command | Description

Reviewer: Should the "full text indexes" and "vector indexes" not be subsections of "indexes for search performance"? Everything else is sort of grouped into subsections 
![Screenshot 2023-12-04 at 10 32 01](https://github.com/neo4j/docs-cypher/assets/7661319/8b141dcb-6a73-4ae7-af33-fe3e75ee51b5)

(Probably not something for this PR, just something I noticed)

Author: They aren't actually `indexes for search performance`, all three should be under a single header though (and Jens is working on a restructuring of the index section so it'll hopefully be nice once done :D)

====

A vector index is a single-label, single-property index for nodes.
A vector index needs to be configured with both the dimensionality of the vector (`INTEGER` between `1` and `2048` _inclusive_), and the measure of similarity between two vectors (case-insensitive `STRING`).
For details, see xref:#indexes-vector-similarity[].

[NOTE]
====
More details about the syntax descriptions can be found link:{neo4j-docs-base-uri}/operations-manual/{page-version}/database-administration/syntax/#administration-syntax-reading[here].
====

.Syntax for creating vector indexes
[options="header", width="100%", cols="5a, 3"]
|===
| Command | Description

| [source, syntax, role=noplay, indent=0]
----
CREATE VECTOR INDEX [index_name] [IF NOT EXISTS]
FOR (n:LabelName)
ON (n.propertyName)
OPTIONS "{" option: value[, ...] "}"
----
| Create a vector index on nodes.

> The options map is mandatory since there are mandatory index configuration settings, the vector dimensions and similarity function.

Reviewer: Consider putting the mandatory options into the example syntax

Author: If feels odd to only add the config as it seems to imply that is the only available options, and with only one existing provider it feels odd to add the provider in the syntax description :( This is also the general way we describe options for all indexes/other places that have them so being more explicit here feels odd as well :(

Reviewer: Fair 🤷

> The options map is mandatory since there are mandatory index configuration settings, the vector dimensions and similarity function.

Reviewer 2: ```suggestion
The options map is mandatory because setting the vector dimensions and similarity function is mandatory when creating a vector index.
```

|===

It is considered best practice to give the index a name when it is created. This name is needed for both dropping and querying the index.
If the index is not explicitly named, it will get an auto-generated name.

[IMPORTANT]
====
The index name must be unique among all indexes and constraints.
====

The `CREATE VECTOR INDEX` command take an `OPTIONS` clause. This has two parts, the `indexProvider` and `indexConfig`.
The provider can only have the default value, `'vector-1.0'`.

> The `indexConfig` is a map from string to string and integers, and is used to set the index-specific configuration settings, the vector dimensions and similarity function.

Reviewer: Consider using an example instead, similar to [what we did for cdc](https://development.neo4j.dev/docs/cdc/current/procedures/output-schema/#_node_changes). That way the users can just copy paste the example & don't have to struggle to get the `[{"` in the right order.

Author: there is an example on creating the index further down though (line 172-177). 

I still think I want to keep the syntax descriptions and not only have examples.

I currently don't remember exactly how the restructuring of the sections will go but it might be that we have one page for all syntax descriptions (like we have for the search performance ones now) and not have fulltext and vector be separate 🤔

> The `indexConfig` is a map from string to string and integers, and is used to set the index-specific configuration settings, the vector dimensions and similarity function.

Reviewer 2: ```suggestion
The `indexConfig` is a `MAP` from `STRING` values to `STRING` and `INTEGER` values, and is used to set the index-specific configuration settings (the vector dimensions and similarity function).
```

|===
| Known issues | Fixed in

| There is no Cypher syntax for creating a vector index.

[TIP]
====

> Use procedure to create the vector index.

Reviewer 2: ```suggestion
Use the procedure `db.index.vector.createNodeIndex` procedure to create the a vector index.
```

Author: ```suggestion
Use the procedure `db.index.vector.createNodeIndex` to create the a vector index.
```
or 
```suggestion
Use the `db.index.vector.createNodeIndex` procedure to create the a vector index.
```
but not double procedure XD

Reviewer 2: Sorry 😃 -  I prefer the first ("use the procedure...")

Reviewer 2: Sorry 😃 - I prefer the first ("Use the procedure...")
